### PR TITLE
[DNM] Experiment with how code gen works for kotlin-client lang

### PR DESCRIPTION
**Pull request by leojkwan**
## What
This is the output directory we get if we run swagger code-gen with the 'kotlin-client' lang config.
```
swagger-codegen generate -i https://api.swaggerhub.com/apis/superfit/superfit-api/1.0.0-oas3 -l kotlin-client -o ./codegen/catalog-api
```

@sufeiiz what do you think of this? With better swagger docs, I think this can save you a ton of time but I know there are things to consider. 

For example, when running this with Swift 4, I'm not sure about the best practice to import the code generated files. Additionally, the code gen models conflicted with my existing models, although I know we can either omit models, are map their swagger model names to something else. The remote object structs on iOS are suffixed ...Remote.swift, so I'd probably need to rename `CatalogSection` from swagger to `CatalogSectionRemote`.

### CODE REVIEW

File: codegen/catalog-api/docs/CatalogSeries.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# CatalogSeries
+
+## Properties
+Name | Type | Description | Notes
+------------ | ------------- | ------------- | -------------
+**id** | [**kotlin.Long**](.md) |  |  [optional]
+**title** | [**kotlin.String**](.md) |  |  [optional]
+**description** | [**kotlin.String**](.md) |  |  [optional]
+**revisionDate** | [**java.time.LocalDateTime**](java.time.LocalDateTime.md) |  |  [optional]
```
Comment (sufeiiz) [inline codegen/catalog-api/docs/CatalogSeries.md]: Currently have this as Strings, but we can convert to LocalDateTime

File: codegen/catalog-api/docs/ProgramCatalogCatalogSeries.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# ProgramCatalogCatalogSeries
+
+## Properties
+Name | Type | Description | Notes
+------------ | ------------- | ------------- | -------------
+**id** | [**java.util.UUID**](java.util.UUID.md) |  |  [optional]
```
Comment (sufeiiz) [inline codegen/catalog-api/docs/ProgramCatalogCatalogSeries.md]: UUID? should all ids be consistent or is this a special case

File: codegen/catalog-api/docs/CatalogSeries.md (hunk 1 above)
Comment (leojkwan) [reply codegen/catalog-api/docs/CatalogSeries.md]: sweet. Is LocalDateTime the standard type for Dates? Is there just a `Date`?

Comment (sufeiiz) [reply codegen/catalog-api/docs/CatalogSeries.md]: https://www.journaldev.com/2800/java-8-date-localdate-localdatetime-instant

File: codegen/catalog-api/src/main/kotlin/io/swagger/client/models/ProgramCatalog.kt [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+ * @param level enum of \"Beginner\", \"Intermediate\", \"Advanced\"
+ * @param season 
+ * @param revisionDate 
+ * @param publishDate 
+ * @param creationDate 
+ * @param availability 
+ * @param mainImageUrl 
+ * @param isLive 
+ */
+data class ProgramCatalog (
```
Comment (sufeiiz) [inline codegen/catalog-api/src/main/kotlin/io/swagger/client/models/ProgramCatalog.kt]: Having this kt class would be a duplicate. if these files are meant to be templates, we shouldnt keep any of the actual data classes. I like the md classes, but maybe remove the kt classes and I can generate them when needed?

Comment (leojkwan) [reply codegen/catalog-api/src/main/kotlin/io/swagger/client/models/ProgramCatalog.kt]: Yeah currently, it's a duplicate of the current class you already wrote out, but ideally we use the generated data class and extend the model with conveniences etc, in a separate file, if necessary.

For example, on iOS, I already have a `CatalogSection` class, which serves as the persistent `Entity` model. I would rename set configure SwaggerCodeGen to map `CatalogSection` to `CatalogSectionRemote`, with the `remote` suffix representing the remote model types from the server.

Comment (leojkwan) [reply codegen/catalog-api/src/main/kotlin/io/swagger/client/models/ProgramCatalog.kt]: btw, this branch is not meant to merge, with what you pointed out as one of the reasons!

**Pull request closed without merging**